File: Reverse String.py
# time: O(n)
# extra-space: O(1)

class Solution(object):
    def reverseString(self, s):
        """
        :type s: List[str]
        :rtype: None Do not return anything, modify s in-place instead.
        """
        n = len(s)
        for i in range(0, n//2):
            s[i], s[n-1-i] = s[n-1-i], s[i]
            
        

File: reverse_string.cpp
// method 1;

class Solution {
public:
    void reverseString(vector<char>& s) {
        int i = 0;
        int j = s.size()-1;
        while(i<=j)
        {
            char temp = s[i];
            s[i] = s[j];
            s[j] = temp;
            i++;
            j--;
        }
    }
};

///////////////////////////////////////////////////////////////////////////////////////////////////////

// method 2 : 

class Solution {
public:
    void reverseString(vector<char>& s) {
        int i = 0;
        int j = s.size()-1;
        while(i<=j)
        {
            swap(s[i++], s[j--]);
        }
    }
};